logstreams/src/main/java/io/zeebe/logstreams/impl/log/index/LogBlockIndex.java

```diff
@@ -152,6 +152,43 @@ public void addBlock(
     lastVirtualPosition = blockPosition;
   }
 
+  /**
+   * Deletes mappings up to {@code deletePosition} (including), with the exception of the last entry
+   * in the index, which will not be deleted. Therefore, this method should be used solely as a
+   * best-effort attempt to free-up disk space and not as a dependable delete operation.
+   *
+   * @param indexContext the log block index context
+   * @param deletePosition the position up to which entries will be deleted
+   */
+  public void deleteUpToPosition(
+      final LogBlockIndexContext indexContext, final long deletePosition) {
+    final AtomicLong lastBlockPosition = new AtomicLong(VALUE_NOT_FOUND);
+
+    indexColumnFamily.whileTrue(
+        indexContext.getDbContext(),
+        (key, val) -> {
+          final long storedBlockPosition = key.getValue();
+
+          if (storedBlockPosition <= deletePosition) {
```

Comment:
Why not just `<` and directly delete the entry? Why it is necessary to store the lastBlockPosition?

Reply:
Because we don't know what entries are in the block that starts at storedBlockPosition, it could be that some of the block's entries are below the delete position and others are not (in which case, you wouldn't delete the block, to avoid losing those mappings). We only know that we can delete a block mapping if we see that the position to delete is after it, which only happens in the next iteration.

Reply:
You're right